packages/types/src/extrinsic/v4/Extrinsic.ts

```diff
@@ -24,6 +25,8 @@ export interface ExtrinsicValueV4 {
  * The third generation of compact extrinsics
  */
 export class GenericExtrinsicV4 extends Struct implements IExtrinsicImpl {
+  #hashCache?: CodecHash;
```

Comment:
This should only be needed on the upper layer instead of duplicating here? (Effectively the Extrinsic is an Enum, so the V4 structure should be be stand-alone)

Reply:
I was not sure how inheritance works here, there was no clear path that this one extends any other ones, it seems to directly extend `Struct`, not `BaseExtrinsic`, so I ended up adding it to both. I can remove it from here if it is indeed covered by the base extrinsic.

Reply:
Will bring it in as-is and cleanup. (It only needs to be on the base, i.e. type `Extrinsic` is created, but that acts like an enum over the V4)